### PR TITLE
Feature/prefix separation

### PR DESCRIPTION
**Pull request by asmodee-bellanie**
Isolates the prefix treatment in its own service.

### CODE REVIEW

File: src/services/message-manager/message-manager.service.ts [hunk 1]
```diff
@@ -23,24 +23,23 @@ export class MessageManagerService {
...
     if (message.guild) {
       if (message.author.bot) return
-      
-      for (const prefix of Object.values(BotPrefixesEnum)) {
-        if (message.content.startsWith(prefix)) {
-          message.channel.startTyping()
 
-          const args = message.content.slice(prefix.length).split(/ +/)
-          const command = args.shift()
+      const prefix: string = PrefixManagerService.getInstance().setPrefix(message.content)
```
Comment (C0ZEN) [inline src/services/message-manager/message-manager.service.ts]: I think that this is strange to have a method called "setX" returning a value.

File: src/services/prefix-manager/prefix-manager.service.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    return PrefixManagerService._instance
+  }
+
+  private _getPrefix(message: string): string {
+    return message.split(/ +/)[0]
+  }
+
+  public setPrefix(message: string): string {
+    for (const prefix of Object.values(BotPrefixesEnum)) {
+      if (prefix === this._getPrefix(message)) return `${prefix} `
```
Comment (C0ZEN) [inline src/services/prefix-manager/prefix-manager.service.ts:20]: Is it tested (not with jest, I mean on Discord)?

**Review by C0ZEN — CHANGES_REQUESTED** [review]
Added comments

File: src/services/prefix-manager/prefix-manager.service.ts (hunk 2 above)
Comment (asmodee-bellanie) [reply src/services/prefix-manager/prefix-manager.service.ts:20]: Yes, it was tested! It works wonders! ^^

File: src/services/message-manager/message-manager.service.ts (hunk 1 above)
Comment (asmodee-bellanie) [reply src/services/message-manager/message-manager.service.ts]: Ok, I will rename it 'getPrefix' :smile:

**Review by C0ZEN — APPROVED** [review]

**Pull request merged**